sampler.go

```diff
@@ -438,7 +477,23 @@ func (s *RemotelyControlledSampler) updateSampler() {
 }
 
 func (s *RemotelyControlledSampler) extractSampler(res *sampling.SamplingStrategyResponse) (Sampler, error) {
-	// TODO create adaptive sampler
+	if strategies := res.GetOperationSamplingStrategies(); strategies != nil {
+		samplingRates := make(map[string]float64)
+		for _, strategy := range strategies {
+			samplingRates[strategy.Operation] = strategy.ProbabilisticSampling.SamplingRate
+		}
+		if sampler, ok := s.sampler.(*adaptiveSampler); ok {
+			s.Lock()
+			sampler.update(defaultLowerBound, defaultSamplingProbability, samplingRates)
+			s.Unlock()
```

Comment:
The issue with returning a new adaptive sampler here is that we incur the costs of building it which we can avoid. I guess I could build a husk of an adaptive sampler (ie just has it's internal fields set but no initialized composite samplers) and change the API to updateSampler(sampler) which all the samplers can implement.

Maybe I'm overthinking things again and we're ok with incurring the building of the adaptive sampler? I'm scared about rtapi clients which will have a ton of operations.